yaksh/views.py

```diff
@@ -1710,9 +1710,12 @@ def grade_user(request, quiz_id=None, user_id=None, attempt_number=None,
 @login_required
 @has_profile
 @email_verified
-def view_profile(request):
+def view_profile(request, user_id=None):
     """ view moderators and users profile """
-    user = request.user
+    if user_id:
```

Comment:
Since any body can brute their way and check other peoples profiles, which was not originally intended, please add the following,

- Add a check to make sure that only Moderators can see other peoples profile.
- Query should be made with username instead of user_ids